src/boundarymps.jl

```diff
@@ -0,0 +1,35 @@
+#GIVEN AN ITENSOR NETWORK CORRESPONDING TO A Lx*Ly grid indexed as (i,j) then perform boundary sweeping using a max_dim of chi_max
+function boundary_PEPS_contractor(
```

Comment:
Here is a simpler implementation in terms of MPO-MPS contraction:
```julia
function contract_boundary_mps(tn::ITensorNetwork; kwargs...)
  dims = maximum(vertices(tn))
  d1, d2 = dims
  vL = MPS([tn[i1, 1] for i1 in 1:d1])
  for i2 in 2:(d2 - 2)
    T = MPO([tn[i1, i2] for i1 in 1:d1])
    vL = contract(T, vL; kwargs...)
  end
  T = MPO([tn[i1, d2 - 1] for i1 in 1:d1])
  vR = MPS([tn[i1, d2] for i1 in 1:d1])
  return inner(dag(vL), T, vR)[]
end
```
Note here I am deducing the size of the grid from the vertices, which is a bit fickle but I think is ok. This version allows more flexibility for choosing different options for the MPO-MPS contraction backend.

There is an issue that for larger networks, in general the norm will either diverge or decay exponentially with the number of tensors in the network, so we will have to be careful about that. A solution is to normalize `vL` as the contraction proceeds and store the norms. A related issue is that in general we probably want to return the boundary MPS `vL` since they will be used to compute expectation values.

Reply:
Okay nice. Yes I agree for now we should just assume the itensornetwork has been constructed via grid((lx,ly)) but could always add in a more rigorous check later on.

Reply:
Yes, we could just add a function `is_grid` which checks the vertices and then a function `grid_size` to get the size of the grid.